crates/test-utils/src/rpc.rs

```diff
@@ -1,73 +1,38 @@
-const ENV_HTTP_ARCHIVE_ENDPOINTS: &str = "HTTP_ARCHIVE_URLS";
-
-// List of general purpose infura keys to rotate through
-static INFURA_KEYS: LazyLock<Vec<&'static str>> = LazyLock::new(|| {
-    let mut keys = vec![
-        // "6cb19d07ca2d44f59befd61563b1037b",
-        // "6d46c0cca653407b861f3f93f7b0236a",
-        // "69a36846dec146e3a2898429be60be85",
-        // "16a8be88795540b9b3903d8de0f7baa5",
-        // "f4a0bdad42674adab5fc0ac077ffab2b",
-        // "5c812e02193c4ba793f8c214317582bd",
-        "22d07e883da246438e63abad14d088f5",
-        "7031545d5b934be8bbdd12e77565178c",
-        "1156b9dbbfe249929d0cbcd3802683eb",
-        "85476c545f8f4e88915ad2260a225d0d",
-        "a254cdc5a4f14b7e8ae076d3b9fb5ea3",
-    ];
-
-    keys.shuffle(&mut rand::thread_rng());
+// List of public archive reth nodes to use
+static RETH_ARCHIVE_HOSTS: LazyLock<Vec<&'static str>> = LazyLock::new(|| {
+    let mut hosts = vec!["reth-ethereum.ithaca.xyz"];
+    hosts.shuffle(&mut rand::thread_rng());
+    hosts
+});
 
-    keys
+// List of public reth nodes to use (archive and non archive)
+static RETH_HOSTS: LazyLock<Vec<&'static str>> = LazyLock::new(|| {
+    let mut hosts = vec!["reth-ethereum.ithaca.xyz", "reth-ethereum-full.ithaca.xyz"];
+    hosts.shuffle(&mut rand::thread_rng());
+    hosts
 });
 
-// List of alchemy keys for mainnet
-static ALCHEMY_KEYS: LazyLock<Vec<&'static str>> = LazyLock::new(|| {
+// List of general purpose DRPC keys to rotate through
+static DRPC_KEYS: LazyLock<Vec<&'static str>> = LazyLock::new(|| {
     let mut keys = vec![
-        // "ib1f4u1ojm-9lJJypwkeZeG-75TJRB7O",
-        // "7mTtk6IW4DwroGnKmG_bOWri2hyaGYhX",
-        // "GL4M0hfzSYGU5e1_t804HoUDOObWP-FA",
-        // "WV407BEiBmjNJfKo9Uo_55u0z0ITyCOX",
-        // "Ge56dH9siMF4T0whP99sQXOcr2mFs8wZ",
-        // "QC55XC151AgkS3FNtWvz9VZGeu9Xd9lb",
-        // "pwc5rmJhrdoaSEfimoKEmsvOjKSmPDrP",
-        // "A5sZ85MIr4SzCMkT0zXh2eeamGIq3vGL",
-        // "9VWGraLx0tMiSWx05WH-ywgSVmMxs66W",
-        // "U4hsGWgl9lBM1j3jhSgJ4gbjHg2jRwKy",
-        // "K-uNlqYoYCO9cdBHcifwCDAcEjDy1UHL",
-        // "GWdgwabOE2XfBdLp_gIq-q6QHa7DSoag",
-        // "Uz0cF5HCXFtpZlvd9NR7kHxfB_Wdpsx7",
-        // "wWZMf1SOu9lT1GNIJHOX-5WL1MiYXycT",
-        // "HACxy4wNUoD-oLlCq_v5LG0bclLc_DRL",
-        // "_kCjfMjYo8x0rOm6YzmvSI0Qk-c8SO5I",
-        // "kD-M-g5TKb957S3bbOXxXPeMUxm1uTuU",
-        // "jQqqfTOQN_7A6gQEjzRYpVwXzxEBN9aj",
-        // "jGiK5vwDfC3F4r0bqukm-W2GqgdrxdSr",
-        // "Reoz-NZSjWczcAQOeVTz_Ejukb8mAton",
-        // "-DQx9U-heCeTgYsAXwaTurmGytc-0mbR",
-        // "sDNCLu_e99YZRkbWlVHiuM3BQ5uxYCZU",
-        // "M6lfpxTBrywHOvKXOS4yb7cTTpa25ZQ9",
-        // "UK8U_ogrbYB4lQFTGJHHDrbiS4UPnac6",
-        // "Lc7oIGYeL_QvInzI0Wiu_pOZZDEKBrdf",
-        // "UVatYU2Ax0rX6bDiqddeTRDdcCxzdpoE",
-        // "bVjX9v-FpmUhf5R_oHIgwJx2kXvYPRbx",
+        "Agc9NK9-6UzYh-vQDDM80Tv0A5UnBkUR8I3qssvAG40d",
+        "AvTJ1CllUUOHr1SH08xt9XDv7jJ0Bj0R8I3fssvAG40d",
```

Comment:
Hey - I was thinking - is it intended? These keys could leak and be used by other persons

<img width="660" alt="image" src="https://github.com/user-attachments/assets/5f2b603b-ff05-44f1-a020-fb8b0ccdd18e" />

Maybe an IP whitelist could be added in DRPC side? Even if the IP list of GitHub runners is long: https://api.github.com/meta

Or they could be injected via CI secret env. + shared only to team members?

Reply:
yeah, that's fine as they should also be able to be used locally, will eventually switch to public reth nodes shortly